src/components/Search/Modal.tsx

```diff
@@ -0,0 +1,60 @@
+import { useDebouncedSearch } from "@/hooks/useDebouncedSearch";
+import { Search } from "@/types";
+import React from "react";
+import tw from "twin.macro";
+import NoResults from "./NoResults";
+import QueryInput from "./QueryInput";
+import Results from "./Results";
+
+interface Props {
+  closeModal: () => void;
+}
+
+const Modal: React.FC<Props> = ({ closeModal }) => {
+  const searchParams = {
+    limit: 10,
+    attributesToHighlight: ["*"],
+    highlightPreTag: "<span>",
+    highlightPostTag: "</span>",
+  };
+  const { clearResponse, query, setQuery, response } =
+    useDebouncedSearch<Search.Document>(
+      process.env.NEXT_PUBLIC_MEILISEARCH_HOST ?? "",
```

Comment:
Nit: Hmmm, we can use something like `zod` to make it so that the access to the env vars is confirmed to be in the env before referencing it in the front-end.

Reply:
Good idea. I'd like to do this in a separate pull, and switch this over in there.

Reply:
#245